### ISSUE TITLE
x/build/cmd/coordinator: sometimes TryBot builds fail to schedule a machine and get stuck

### ISSUE THREAD

**Issue by dmitshur** [golang/go#55947]
Consider the TryBot run on CL 431956. Most of its builds have finished, with exception of linux-amd64-boringcrypto, linux-amd64-nounified, and linux-amd64-unified which are in waiting_for_machine state upwards of 20+ hrs:

<img width="502" alt="image" src="https://user-images.githubusercontent.com/1924134/193052289-f918ecac-cb8f-4fef-a70c-13bbe34d1b00.png">

<details>

<img width="881" alt="image" src="https://user-images.githubusercontent.com/1924134/193051969-4cc6043e-30cb-4c64-ac37-cd74064a4023.png">

</details>

Those TryBot runs may have been started at a busy time, but they aren't being scheduled even after the GCE pool capacity is available.

```
GCE pool capacity: 3/20000 instances; 0/10000 CPUs, 24/200 C2_CPUS, 0/800 N2_CPUS, 0/800 N2D_CPUS
```

<details>

<img width="772" alt="image" src="https://user-images.githubusercontent.com/1924134/193053041-500c041d-9ee4-43c2-b2c8-e222c0d38ecd.png">

<img width="676" alt="image" src="https://user-images.githubusercontent.com/1924134/193053140-e17cdfed-4ba5-47c9-a653-382939ac140d.png">

</details>

This is a problem for TryBot speed (meta tracking issue #17104).

CC @golang/release, @thanm, @cherrymui.

**Comment by thanm** [comment]
Nice detective work. Infinite loop in checkDep (or whatever it is) looks more actionable than general slowness...

**Comment by dmitshur** [comment]
We should wait after CL 435496 is deployed and see if this happens again. It's possible that CL is all that's needed to fix this problem.

**Comment by dmitshur** [comment]
We have learned that the overall problem of quota exhaustion still happens after coordinator restarts during a busy time; @prattmic and @heschi have made more progress on understanding that, but it's not the problem that this issue was tracking.

I suspect this particular instance was really a rare case of `checkDep` stalling, and I haven't seen it happen again after [CL 435496](https://go.dev/cl/435496). Closing for now.
